Review for Fritz

fritzo opened the pull request.

fritzo commented on puddle-editor/source/lib/menu.js:
```diff
@@ -0,0 +1,260 @@
+        var varName = syntax.tree.getFresh(term);
+        var fresh = VAR(varName);
+
+        if (name === 'ASSERT') {
```
There is a lot of duplicated code copied from `var build = funciont () {...}`. Refactor?

yangit replied:
Sure, I was half way into factoring it out as a separate module.

fritzo commented on puddle-server/server/server.js:
```diff
@@ -0,0 +1,94 @@
+    var state = puddleSocket.getState();
+    var lines = [];
+    var ids = [];
+    //TODO fix inconsistency between Analyst, puddle-hub and puddle-syntax API
```
Let's discuss strings vs lines...

fritzo commented on puddle-editor/source/lib/puddle-syntax-0.1.2/node_modules/underscore/underscore.js:
```diff
@@ -0,0 +1,1415 @@
+//     Underscore.js 1.7.0
```
Will we remove underscore from the source tree when we remove puddle-syntax 0.1.2?

yangit replied:
Yes we are not supposed to keep puddle-syntax 0.1.2 there for long.

fritzo commented:
@yangit I have left a few comment on this pull request.

yangit committed:
Editor does not know about menu
Menu API simplified
Renamed navigator => menuRenderer
Menu does not know about menuRenderer

yangit committed:
* editor does not care about view or menu
* old puddle-syntax removed
* view and menu code refactored
* keycode replaced with keypressJS lib

yangit committed:
This is a big change to client side libraries API.
Editor.js is now split among cursor.js, menu-builder.js, forest.js
Event listener is used across libraries to notify each other of events.
See reference.md for better description.

yangit commented:
Please review.
Most important info is in ./doc/reference.md

fritzo commented on puddle-editor/source/app/corpus.js:
```diff
@@ -0,0 +1,382 @@
+        var linesForAnalyst = _.map(lines, function (line, id) {
+            ids.push(id);
+            //TODO why line.name is undefined after syntax.compiler.dumpLine?
+            //Should not dumpLine be compatible with analyst?
```
Agreed, `syntax.compiler.dumpLine` should be compatible with analyst.

fritzo commented on puddle-editor/source/app/corpus.js:
```diff
@@ -0,0 +1,382 @@
+    var ASSERT = syntax.compiler.fragments.church.ASSERT;
+    var churchTerm = ASSERT(HOLE);
+    var line = syntax.compiler.dumpLine(churchTerm);
+    line.name = null;
```
this line will be obsolete when `syntax.compiler.dumpLine` is updated.

fritzo commented:
+1

yangit committed:
Fix rendering issue #20

yangit committed:
Fix #18, puddle-hub too verbose in server console.

yangit committed:
Fix #17 Issue
Refactor the way cursor is injected into the term and removed from term.
Cursor injection simplified in view.js

yangit committed:
Fix #19
Added line checkout notification across multiple clients.

yangit commented:
Travis is not passing until puddle-syntax version in NPM is bumped
